sdr/concepts_and_interactions.md

```diff
@@ -0,0 +1,202 @@
+* Collection - _a collection of items, belonging to an APO_
+* Item - _an item can have files; it belongs to a (one or more?) collection_
+
+## Object Registration and Accessioning
+
+How people get objects get into SDR
+
+### Argo
+
+* E.g. "Use Argo to create an object without any files" - https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/create_object_no_files_spec.rb
+* E.g. Create an object, version it, confirm it was preserved - https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/create_preassembly_image_spec.rb
+  * touches on accessioning, preservation, and the event log
+
+### H2
+
+Use H2 to create an object - https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/create_object_h2_spec.rb
+
+## cocina-models
+_SDR data model written as syntactically validatable with dry-struct, dry-types and openapi_
+
+(ask JCoyne or JLitt)
+
+## Workflows and Robots
+
+Robots are what we call our individual accessioning processing steps which are grouped into "workflows", coordinated by the workflow server (and Resque and resque-pool).  They do things like updating the SDR metadata store (currently Fedora data streams), generating technical metadata, handing off to the preservation system a copy of each object version, etc.  
+
+Together, workflow server and the robots provide a system for managing the SDR accessioning pipeline: the robots ingest content into the SDR; workflow server coordinates by queuing tasks to a shared Redis instance.
+
+The robots inherit their worker functionality from [the lybercore gem](https://github.com/sul-dlss/lyber-core/).
+
+Each type of robot has one or more VMs of its own, but all share the same Redis instance, and all are managed on their respective VMs by resque-pool.
+
+This architecture was chosen before open source workflow automation tools like Airflow were available/mature.
+
+Another thing that adds indirection/confusion: sometimes the robots will perform the heavy lifting of a task, but sometimes even they call out to other services.  For example, in the `preservationIngestWF`, the `validate-moab` step in the preservation_robots actually makes a REST call to preservation_catalog to do the validation asynchronously (because that service will do all auditing and cloud replication after the robot finishes updating the content).  And then preservation_catalog tells workflow service when it's done and workflow service tells the preservation_robots to do the next step.
+
+(ask anyone and hand waving will occur.  Perhaps Peter? Andrew? Naomi? JCoyne? - we can re-figure out the layers of legacy code as a team)
+
+
+## Access Rights for digital content
+
+(ask JCoyne, John)
+
+* What's an APO?
+* Where are access restrictions computed when interacting with a specific object? - various codebases use the logic provided by the dor-rights-auth gem:  https://github.com/sul-dlss/dor-rights-auth
```

Comment:
I don't think that dor-rights-auth is used by any infrastructure projects. It is a transitive dependency of DSA, but I don't think it's used.

Reply:
I think it may be.

Reply:
ah yeah, i was not sure about this, will ticket figuring it out.

Reply:
https://github.com/sul-dlss/infrastructure-team/issues/7